core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -321,6 +325,48 @@ protected function compile()
 					$objTemplate->hasContext = true;
 				}
 
+				// Add image to search results
+				$objTemplate->hasImage = false;
+				if (isset($arrResult[$i]['meta']) && null !== $arrResult[$i]['meta'])
+				{
+					$meta = json_decode($arrResult[$i]['meta'], true);
+					// Reverse so the last element wins
+					$meta = array_reverse($meta);
+					foreach ($meta as $v)
+					{
+						if (!isset($v['https://schema.org/primaryImageOfPage']))
+						{
+							continue;
+						}
+
+						if ($figureBuilder = $this->getFigureBuilderIfResourceExists($v['https://schema.org/primaryImageOfPage']['contentUrl']))
+						{
+							$objTemplate->hasImage = true;
+
+							$figureMeta = array();
+							if (isset($v['https://schema.org/primaryImageOfPage']['caption']))
+							{
+								$figureMeta[Metadata::VALUE_CAPTION]  = $v['https://schema.org/primaryImageOfPage']['caption'];
+							}
+							if (isset($v['https://schema.org/primaryImageOfPage']['name']))
+							{
+								$figureMeta[Metadata::VALUE_TITLE]  = $v['https://schema.org/primaryImageOfPage']['name'];
+							}
+
+							if (isset($v['https://schema.org/primaryImageOfPage']['alternateName']))
+							{
+								$figureMeta[Metadata::VALUE_ALT]  = $v['https://schema.org/primaryImageOfPage']['alternateName'];
+							}
```

Comment:
How about this? Might be a bit easier to read:

```suggestion
							$figureMeta = new Metadata(array_filter([
								Metadata::VALUE_CAPTION => $v['https://schema.org/primaryImageOfPage']['caption'] ?? null,
								Metadata::VALUE_TITLE => $v['https://schema.org/primaryImageOfPage']['name'] ?? null,
								Metadata::VALUE_ALT => $v['https://schema.org/primaryImageOfPage']['alternateName'] ?? null,
							]));
```

Reply:
Nice, that's done too.